ui/app/src/Updates.elm

```diff
@@ -96,6 +99,64 @@ update msg model =
             in
                 ( { model | filter = { filter | text = t } }, Cmd.none )
 
+        KeyDownMsg code ->
+            let
+                mode =
+                    if List.member code modifierKeys then
+                        Append
+                    else
+                        Replace
+            in
+                ( { model | mode = mode }, Cmd.none )
+
+        KeyUpMsg code ->
+            let
+                mode =
+                    if List.member code modifierKeys then
+                        Replace
+                    else
+                        Append
+            in
+                ( { model | mode = mode }, Cmd.none )
+
+        AddLabel msg ( key, value ) ->
+            let
+                filter =
+                    model.filter
+
+                label =
+                    key ++ "=" ++ (toString value)
+
+                labels =
+                    case model.mode of
+                        Append ->
+                            Maybe.withDefault "" filter.text
+                                |> Regex.replace Regex.All (Regex.regex "{|}") (\_ -> "")
+                                |> (\cleaned ->
+                                        if String.isEmpty cleaned then
+                                            []
+                                        else
+                                            Regex.split Regex.All (Regex.regex "\\s*,\\s*") cleaned
```

Comment:
I check in the other if branch if it is an empty string and return an empty list